pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/generator/ParametersDirective.java

```diff
@@ -81,11 +86,23 @@ public String getDisplayName() {
         public String toGroovy(@NonNull ParametersDirective directive) {
             StringBuilder result = new StringBuilder("parameters {\n");
             for (ParameterDefinition param : directive.parameters) {
-                result.append(Snippetizer.object2Groovy(UninstantiatedDescribable.from(param)));
+                result.append(Snippetizer.object2Groovy(customUninstantiate(param)));
```

Comment:
This is the fix for the directive generator. I wanted to add a test, but the test harness assumes that things round-trip, and as far as I know the directive generator is one-way only. Maybe things could have been tweaked to get the test passing by adding some instantiation-related code here that would only be used only in tests, but I'm not sure.